File: app/src/main/java/grey/smarthouse/ui/activities/SingleFragmentActivity.java
package grey.smarthouse.ui.activities;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.appcompat.app.AppCompatActivity;
import android.view.View;
import android.widget.ProgressBar;


import grey.smarthouse.R;

/**
 * Created by GREY on 17.05.2018.
 */

public abstract class SingleFragmentActivity extends AppCompatActivity {

    protected abstract Fragment createFragment();


    ProgressBar PBar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_default);
        PBar = (ProgressBar) findViewById(R.id.progressRelaySettings);
        }

    protected void addFragment() {
        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(R.id.fragment_container);
        if (fragment == null) {
            fragment = createFragment();
            fm.beginTransaction()
                    .add(R.id.fragment_container, fragment)
                    .commit();
        }
        PBar.setVisibility(View.INVISIBLE);
    }

}

File: app/src/main/java/grey/smarthouse/ui/SingleFragmentActivity.kt
package grey.smarthouse.ui

import android.os.Bundle
import android.view.View
import android.widget.ProgressBar
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import grey.smarthouse.R

/**
 * Created by GREY on 17.05.2018.
 */

abstract class SingleFragmentActivity : AppCompatActivity() {

    protected abstract fun createFragment(): Fragment?
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_default)
    }

    protected fun addFragment() {
        val fm = supportFragmentManager
        var fragment = fm.findFragmentById(R.id.fragment_container)
        if (fragment == null) {
            fragment = createFragment()
            fragment?.let {
                fm.beginTransaction()
                        .add(R.id.fragment_container, it)
                        .commit()
            }
        }
    }

}
